pilot/cmd/sidecar-injector/main.go

```diff
@@ -131,25 +133,56 @@ func patchCertLoop() error {
 	if err != nil {
 		return err
 	}
-
 	watchDir, _ := filepath.Split(flags.caCertFile)
 	if err = watcher.Watch(watchDir); err != nil {
 		return fmt.Errorf("could not watch %v: %v", flags.caCertFile, err)
 	}
 
+	if err = util.PatchMutatingWebhookConfig(client.AdmissionregistrationV1beta1().MutatingWebhookConfigurations(),
+		flags.webhookConfigName, flags.webhookName, caCertPem); err != nil {
+		return err
+	}
+
+	shouldPatch := make(chan struct{})
+
+	watchlist := cache.NewListWatchFromClient(
+		client.AdmissionregistrationV1beta1().RESTClient(),
+		"mutatingwebhookconfigurations",
+		"",
+		fields.ParseSelectorOrDie(fmt.Sprintf("metadata.name=%s", flags.webhookConfigName)))
+
+	_, controller := cache.NewInformer(
+		watchlist,
+		&v1beta1.MutatingWebhookConfiguration{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				config := newObj.(*v1beta1.MutatingWebhookConfiguration)
+				for i, w := range config.Webhooks {
+					if w.Name == flags.webhookName {
+						if !bytes.Equal(config.Webhooks[i].ClientConfig.CABundle, caCertPem) {
```

Comment:
nit: you can dispense with this extra if.

Reply:
@ostromart why? In theory any other field of the MutatingWebhookConfiguration could be changed, so, this makes sure we only patch the cert if it has changed. Otherwise this entire for loop could be removed...

Reply:
I just meant you could write:
if w.Name == flags.webhookName &&
 !bytes.Equal(config.Webhooks[i].ClientConfig.CABundle, caCertPem) {
to avoid the extra indentation.

Reply:
@ostromart ah, gotcha, updated, thanks.